sources/HUBCollectionView.m

```diff
@@ -42,6 +46,27 @@ - (void)setContentOffset:(CGPoint)contentOffset
     [super setContentOffset:contentOffset];
 }
 
+- (NSMutableSet<NSString *> *)registeredCollectionViewCellReuseIdentifiers
+{
+    if (!_registeredCollectionViewCellReuseIdentifiers) {
+        _registeredCollectionViewCellReuseIdentifiers = [NSMutableSet new];
+    }
+
+    return _registeredCollectionViewCellReuseIdentifiers;
+}
+
+- (__kindof UICollectionViewCell *)dequeueReusableCellWithReuseIdentifier:(NSString *)identifier
+                                                             forIndexPath:(NSIndexPath *)indexPath
+                                                cellClassWhenUnregistered:(Class)cellClass
+{
+    if (![self.registeredCollectionViewCellReuseIdentifiers containsObject:identifier]) {
+        [self registerClass:cellClass forCellWithReuseIdentifier:identifier];
```

Comment:
How about doing this in the `loadCollectionViewIfNeeded` method? Would also allow us to not expose any new API. Or is there a specific reason we want to do it this way (i.e. JIT for when the cell is being dequeued)? Feels a bit scary and on the edge of the API constraints.